crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -385,12 +444,48 @@ impl TryFrom<StoredObject> for Object {
 
         Ok(Self {
             address,
+            state: ObjectState::Active(native_object),
             stored: Some(stored_object),
-            native: native_object,
         })
     }
 }
 
+impl TryFrom<StoredHistoryObject> for Object {
+    type Error = Error;
+
+    fn try_from(history_object: StoredHistoryObject) -> Result<Self, Error> {
+        let address = addr(&history_object.object_id)?;
+        if history_object.object_status == NativeObjectStatus::Active as i16 {
```

Comment:
Can this be written using `match`?

Reply:
I could do something like:
```
        match history_object.object_status {
            x if x == NativeObjectStatus::Active as i16 => {
```
but i feel like that defeats the purpose of using match over if

it's because history_object.object_status is i16 while ObjectStatus is a different type (an enum of variants Active=0, WrappedOrDeleted=1)

Reply:
haha yes of course, i'll stop being lazy :P